src/main/scala/scorex/core/utils/NetworkTime.scala

```diff
@@ -9,31 +9,28 @@ import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.duration._
 import scala.util.Left
 
-protected trait NetworkTime extends ScorexLogging {
+object NetworkTime {
+  def localWithOffset(offset: Long): Long = System.currentTimeMillis() + offset
 
-  val offset: NetworkTime.Offset
-  val lastUpdate: NetworkTime.Time
-
-  protected def localWithOffset() = System.currentTimeMillis() + offset
-}
-
-object NetworkTime extends ScorexLogging {
   type Offset = Long
   type Time = Long
+}
 
-  private type State = Either[(NetworkTime, Future[NetworkTime]), NetworkTime]
+protected case class NetworkTime(offset: NetworkTime.Offset, lastUpdate: NetworkTime.Time)
+
+case class NtpTimeProviderSettings(server: String, updateEvery: FiniteDuration, timeout: FiniteDuration)
 
-  private val TimeTillUpdate = 1000 * 60 * 30L
-  // 30 minutes
-  private val NtpServer = "pool.ntp.org"
+class NtpTimeProvider(ntpSettings: NtpTimeProviderSettings) extends ScorexLogging {
```

Comment:
What does "Ntp" mean? Does it mean "NetworkTimeProvider"? If so, I would suggest renaming "NtpTimeProvider" to "NetworkTimeProvider", to avoid redundancy in the name.